bert/README.md

```diff
@@ -0,0 +1,244 @@
+
+So, for example, if you want to use ALiBi positional embeddings (and [you probably should](https://ofir.io/The-Use-Case-for-Relative-Position-Embeddings/), they seem to improve LM perplexity, downstream accuracy, and allow the model to generalize to longer sequences than seen at train time), you need to train from scratch (or fine-tune from a checkpoint which was pre-trained with ALiBi positional embeddings, which we will be releasing!). Or if you want to use Gated Linear Units in your feedforward layers ([as recommended by Noam Shazeer](https://arxiv.org/abs/2002.05202), one of the authors of the original Transformers paper), again, you have to train with them from scratch.
+
+Another good example is domain-specific tokenization. In the biomedical domain, words may be split by the pre-trained BERT tokenizer in ways that make downstream tasks more difficult and computationally expensive. For example, the common drug "naloxone" in tokenized by `bert-base-uncased` tokenizer into the 4 tokens `[na, ##lo, ##xon, ##e]` [(Gu et al., 2020)](https://arxiv.org/abs/2007.15779), making tasks like NER more difficult and using more of the limited sequence length available.
+
+Now that we've convinced you that you should train a Mosaic BERT from scratch, let's get into the how :) 
+
+# Setup
+
+Here's what you need to get started with our BERT implementation:
+
+* Use a Docker image with PyTorch 1.12+, e.g. [MosaicML's PyTorch base image](https://hub.docker.com/r/mosaicml/pytorch/tags)
+  * Recommended tag: `mosaicml/pytorch:1.12.1_cu116-python3.9-ubuntu20.04`
+  * This image comes pre-configured with the following dependencies:
+    * PyTorch Version: 1.12.1
+    * CUDA Version: 11.6
+    * Python Version: 3.9
+    * Ubuntu Version: 20.04
+
+* Use a system with NVIDIA GPU(s)
+
+* Install requirements via: `pip install -r requirements.txt`
+
+* Prepare a local copy of the C4 dataset via instructions below.
+
+## Dataset preparation
+
+To run pre-training, you'll need to make yourself a local copy of the C4 pre-training dataset.
+If you only want to profile these BERTs, we recommend that you **only download and prepare the `val` split**,
+and use it for both train and eval in your script. Just change `split: train` to `split: val` in your run YAML.
+Alternatively, feel free to substitute our dataloader with one of your own in the script [main.py](./main.py#L101)!
+
+In this benchmark, we train BERTs on the [C4: Colossal, Cleaned, Common Crawl dataset](https://huggingface.co/datasets/c4).
+We first convert the dataset from its native format (a collection of zipped JSONs)
+to MosaicML's streaming dataset format (a collection of binary `.mds` files).
+Once in `.mds` format, we can store the dataset in a central location (filesystem, S3, GCS, etc.)
+and stream the data to any compute cluster, with any number of devices, and any number of CPU workers, and it all ~ just works ~ .
```

Comment:
```suggestion
and stream the data to any compute cluster, with any number of devices, and any number of CPU workers, and it all ~just works~ .
```
Assuming the lack of whitespace doesn't mess with the markdown. Otherwise just rm the tildes.